gwcs/wcs.py

```diff
@@ -634,3 +641,269 @@ def fix_inputs(self, fixed):
         new_pipeline.append((step0[0], new_transform))
         new_pipeline.extend(self.pipeline[1:])
         return self.__class__(new_pipeline)
+
+    def to_fits_sip(self, bounding_box, max_pix_error=0.25, degree=None,
+                    max_inv_pix_error=0.25, inv_degree=None, 
+                    npoints=32, verbose=False):
+        """
+        Construct a SIP-based approximation to the WCS in the form of a FITS header
+
+        This assumes a tangent projection.
+
+        The default mode in using this attempts to achieve roughly 0.25 pixel
+        accuracy over the whole image.
+
+        Parameters
+        ----------
+        bounding_box : a pair of tuples, each consisting of two integers
```

Comment:
OK, that seems odd for array coordinates, but I'll make the change.

Reply:
Well, a long time ago it was decided that (for jwst at least) the bounding_box will define the edges of the image, not the center of the corner pixels and since the center of the pixel is an integer ...